src/comment-npm-publish/index.ts

```diff
@@ -18,16 +18,4 @@ export async function main(): Promise<void> {
   log('Action end');
 }
 
-main()
-  .then(() => {
-    process.stdin.destroy();
-    // eslint-disable-next-line unicorn/no-process-exit
-    process.exit(0);
-  })
-  // eslint-disable-next-line unicorn/prefer-top-level-await
-  .catch((error) => {
-    // eslint-disable-next-line no-console
-    console.log('Action Error - exit 1 - error:', error);
-    // eslint-disable-next-line unicorn/no-process-exit
-    process.exit(1);
-  });
+await main();
```

Comment:
given we're just awaiting main() at the top level of the module, do we need the main function at all, just make everything top level?  this feels very script-y anyway

Reply:
good idea

Reply:
tried that and making everything top level, it works but there is one advantage in having a 'main' function when it comes to testing.
what happens is that we'll have to do dynamic import in the tests; and before the dynamic import we normally setup nocking and environment variables. Jest has its own module catching mechanism, which causes some of the tests start failing. one trick is to do `jest.resetModules()` everywhere.
it seems easier and less surprise if we call the main function instead.  
what do you think? should we make the change, or keep the main?

Reply:
if we're await()ing main, isn't it effectively having everything top-level anyway?  my thought is we move main() into a separate module, and all the index.ts does is import and await it.  that way tests can do their thing without side effects going on at the module level.